inference-engine/src/vpu/graph_transformer/src/middleend/passes/merge_permute_stages.cpp

```diff
@@ -55,12 +55,33 @@ class PassImpl : public Pass {
                                   permuteStage->output(0)->desc().dimsOrder());
     }
 
-    static bool isTrivialPermute(const PermutationIndexVector& permuteDims) {
-        for (size_t i = 0; i < permuteDims.size(); ++i)
-            if (i != permuteDims[i])
-                return false;
+    static bool isTrivialPermute(const PermutationIndexVector& permuteDims, const vpu::DimValues& dims) {
+        auto defDims = DimsOrder::fromNumDims(dims.size());
+        std::vector<int> dimsVector(dims.size());
+        for (const auto& dim : dims) {
+            dimsVector[defDims.dimInd(dim.first)] = dim.second;
+        }
+
+        size_t i;
+        bool swapAdjacentDims, dimIsOne = false;
+
+        for (i = 0; i < permuteDims.size(); ++i)
```

Comment:
Please add braces for `for`

Reply:
Also, if you will fix my objections, than you can define `i` right here.

Reply:
Done